lib/Platform/Intl/PlatformIntlICU.cpp

```diff
@@ -16,19 +17,63 @@ using namespace ::hermes;
 
 namespace hermes {
 namespace platform_intl {
+namespace {
 
+// https://tc39.es/ecma402/#sec-canonicalizelocalelist
+vm::CallResult<std::vector<std::u16string>> canonicalizeLocaleList(
+    vm::Runtime &runtime,
+    const std::vector<std::u16string> &locales) {
+  // 1. If locales is undefined, then a. Return a new empty list
+  if (locales.empty()) {
+    return std::vector<std::u16string>{};
+  }
```

Comment:
I'm about to merge this and just wanted to note two minor things to fix up. 

1. We use `///` for doc-comments, and it is generally preferable to use links to a specific version of the spec (as is done in the Apple version).
2. It looks like there's an extra line here that isn't in the Apple version. Note that given how the function is written, this check is not needed.

```suggestion
/// https://402.ecma-international.org/8.0/#sec-canonicalizelocalelist
vm::CallResult<std::vector<std::u16string>> canonicalizeLocaleList(
    vm::Runtime &runtime,
    const std::vector<std::u16string> &locales) {
  // 1. If locales is undefined, then
  //   a. Return a new empty List
  // Not needed, this validation occurs closer to VM in 'normalizeLocales'.
  // 2. Let seen be a new empty List.
  std::vector<std::u16string> seen;
```